python/cuml/test/test_arima.py

```diff
@@ -0,0 +1,590 @@
+    np.testing.assert_allclose(l2_error_predict1, l2_error_ref1, rtol=rtol)
+    rtol = 1e-3
+    np.testing.assert_allclose(l2_error_forecast0, l2_error_fc_ref0, rtol=rtol)
+    np.testing.assert_allclose(l2_error_forecast1, l2_error_fc_ref1, rtol=rtol)
+    
+
+
+def bench_arima(num_batches=240, plot=False):
+
+    ns = len(t)
+    y_b = np.zeros((ns, num_batches))
+
+    for i in range(num_batches):
+        y_b[:, i] = np.random.normal(size=ns, scale=2000) + data_smooth
+
+    # if plot:
+    #     plt.plot(t, data_smooth, "r--", t, data0, "k-", t, y_b[:, 0], t, y_b[:, 1], t, y_b[:, 2])
+    #     plt.show()
+    # return
+
+    p, d, q = (1, 1, 1)
+    order = (p, d, q)
+
+    x0 = np.array([])
+
+    start = timer()
+
+    mu0, ar0, ma0 = arima.estimate_x0(order, num_batches, y_b)
+
+    batched_model = arima.fit(y_b, order,
+                              mu0,
+                              ar0,
+                              ma0,
+                              opt_disp=-1, h=1e-9)
+
+    end = timer()
+
+    print("GPU Time ({} batches) = {} s".format(num_batches, end - start))
+    print("Solver iterations (max/min/avg): ", np.max(batched_model.niter), np.min(batched_model.niter),
+          np.mean(batched_model.niter))
+
+    yt_b = batched_model.predict_in_sample()
+
+    if plot:
+        plt.plot(t, y_b[:, 0], "k-", t, yt_b[:, 0], "r--", t, data0, "g--", t, data_smooth, "y--")
+        plt.show()
+
+def test_grid_search(num_batches=2):
+    ns = len(t)
+    y_b = np.zeros((ns, num_batches))
+
+    for i in range(num_batches):
+        y_b[:, i] = np.random.normal(size=ns, scale=2000) + data_smooth
+
+    best_model, ic = arima.grid_search(y_b, d=1)
+
+    if num_batches == 2:
+        np.testing.assert_array_equal(best_model.order, [(0, 1, 1), (0, 1, 1)])
+
+
+def demo():
+    num_samples = 200
+    xs = np.linspace(0, 1, num_samples)
+    np.random.seed(12)
+    noise = np.random.normal(scale=0.05, size=num_samples)
+    noise2 = np.random.normal(scale=0.05, size=num_samples)
+    ys1 = noise + 0.5*xs + 0.1*np.sin(xs/np.pi)
+    ys2 = noise2 + 0.25*xs + 0.15*np.sin(0.8*xs/np.pi)
+    ys = np.zeros((num_samples, 2))
+    ys[:, 0] = ys1
+    ys[:, 1] = ys2
+
+    plt.plot(xs, ys1, xs, ys2)
+
+    mu0, ar0, ma0 = arima.estimate_x0((1,1,1), 2, ys)
+    
+    model = arima.fit(ys, (1,1,1), mu0, ar0, ma0)
+
+    yp = model.predict_in_sample()
+    yfc = model.forecast(50)
+    dx = xs[1] - xs[0]
+    xfc = np.linspace(1, 1+50*dx, 50)
+    plt.plot(xs, yp, xfc, yfc)
+
+
+def test_stationarity():
+
+    num_samples = 200
+    xs = np.linspace(0, 1, num_samples)
+    np.random.seed(12)
+    noise = np.random.normal(scale=0.1, size=num_samples)
+    ys1 = noise + 0.5*xs
+    ys2 = noise
+
+    num_batches = 2
+    ys_df = np.zeros((num_samples, num_batches), order="F")
+    ys_df[:, 0] = ys1
+    ys_df[:, 1] = ys2
+
+    d_b = stationarity(ys_df)
+    np.testing.assert_array_equal(d_b, [1, 0])
+
+
+
+if __name__ == "__main__":
```

Comment:
I don't think we'd need an explicit list of tests, if we are using `py.test` to run them.

Reply:
Previously I've just been running the file, or using iPython. Just tried py.test, which works well, so I've removed the list.